src/DataDryad.jl

```diff
@@ -49,6 +49,20 @@ function get_urls(repo::DataDryad, page)
     urls
 end
 
+function get_checksums(repo::DataDryad, page)
+    checksums = []
+    checksum_eles = matchall(sel"a", page.root)
+    for checksum_ele in checksum_eles
+        if ismatch(r"\bresource\/doi:[0-9]*.[0-9]*\/dryad.[a-z, 0-9]*\/[0-9]+\b", checksum_ele.attributes["href"])
+            checksum = match(r"\bresource\/doi:[0-9]*.[0-9]*\/dryad.[a-z, 0-9]*\/[0-9]+\b", checksum_ele.attributes["href"])
+            md5 = (:md5, text_only(getpage(replace(checksum.match, "resource/doi:", "https://datadryad.org/mn/checksum/doi:")).root))
+            info("The generated registration block uses md5 hash, the MD5.jl package must be loaded to run the registration")
+            push!(checksums, md5)
+        end
+    end
+    checksums
+end
```

Comment:
I find this a bit hard to read.

Below is how I would have written this code:
Key points:

 1. Keep line length under 79 characters where possible
 2. Use more variables to express intermediary steps, (aids in 1, and in 3)
 3. Use meaningful variable names as it makes the code easier to understand, the things returned by ,`matchall(sel"a", page.root)`  is all links, not all checksums

For more specific points see comments


```
function get_checksums(repo::DataDryad, page)
    info("The generated registration block uses md5 hash, the MD5.jl package must be loaded to run the registration")
    # only print info once, it will apply to every checksum for DataDryad
    doi_regex = r"\bresource\/doi:[0-9]*.[0-9]*\/dryad.[a-z, 0-9]*\/[0-9]+\b"

    checksums = []
    link_eles = matchall(sel"a", page.root)
    for link_ele in link_eles
        doi_match = match(doi_regex,  link_eles.attributes["href"])
        if doi_match != nothing # match returns `nothing` on non-match
            doi = doi_match.match
            checksum_url = replace(doi, "resource/doi:", "https://datadryad.org/mn/checksum/doi:")
            md5_hash = text_only(getpage(checksum_url).root)
            md5 = (:md5, md5_hash)            
            push!(checksums, md5)
        end
    end
    @assert length(checksums)>0
    checksums
end
```